### ISSUE TITLE
Thumbnails won't generate

### ISSUE THREAD

**Issue by Applepiee** [Piwigo/piwigo-videojs#162]
Hello,

I installed your plugin, however the plugin is unable to generate any thumbnails.
When trying to generate I get the error: "You ask me to do nothing, are you sure?"

I do see 4 empty fields under Program Dependencies, but when I fill them in, there is no save button. Clicking the submit at the bottom gives the same error again (You ask me to do nothing, are you sure?) and causes the fields to be empty again.

How can I troubleshoot this?

**Comment by BillW2016** [comment]
Hi, I have exactly the same problem.  

Videos play just fine in the gallery, but the Synchronize function will not generate thumbnails.  After clicking "submit", it lists all the videos in green, but doesn't do anything except give me the error: "You ask me to do nothing, are your sure?"  'Perform a simulation only' is NOT checked.

Can you please provide some help for this?

**Comment by jensbrak** [comment]
Same here. Part from that, awesome plugin, well done!

I get the following message(s):
```
Warning: array_merge(): Expected parameter 1 to be an array, null given in /var/www/html/piwigo/piwigo/plugins/piwigo-videojs/admin/admin_sync.php on line 53

Notice: Undefined index: output in /var/www/html/piwigo/piwigo/plugins/piwigo-videojs/admin/admin_sync.php on line 66

Warning: array_merge(): Expected parameter 1 to be an array, null given in /var/www/html/piwigo/piwigo/plugins/piwigo-videojs/admin/admin_sync.php on line 78
```

**Comment by ipsedix** [comment]
Hi, probably you guys have to go to MySQL editor and execute this query for table `config`:
UPDATE `config` SET `value`="a:0:{}" WHERE `param`="vjs_sync" 

Parameter `vjs_sync` should be a serialized array, but instead it equals like "N;". It causes warning "array_merge(): Expected parameter 1 to be an array".

**Comment by jensbrak** [comment]
Will try, thanks for info!  That seem like a small thing to try for sure.

Seems this plugin is somewhat abandoned and needs to be updated to work with newer versions of piwigo and other dependencies. Or at least get some love and attention. 

Much appreciated you provided a possible solution, since the plugin is a really neat addition to an awesome application (just discovered it and it fits my purposes perfectly!)

**Comment by cerealconyogurt** [comment]
> Will try, thanks for the info! That seems like a small thing to try for sure.
> 
> Seems this plugin is somewhat abandoned and needs to be updated to work with newer versions of piwigo and other dependencies. Or at least get some love and attention.
> 
> Much appreciated you provided a possible solution since the plugin is a really neat addition to an awesome application (just discovered it and it fits my purposes perfectly!)

Did you test it? I am having the same issue

**Comment by Applepiee** [comment]
> Same here. Part from that, awesome plugin, well done!
> 
> I get the following message(s):
> 
> ```
> Warning: array_merge(): Expected parameter 1 to be an array, null given in /piwigo/plugins/piwigo-videojs/admin/admin_sync.php on line 53
> 
> Notice: Undefined index: output in /piwigo/plugins/piwigo-videojs/admin/admin_sync.php on line 66
> 
> Warning: array_merge(): Expected parameter 1 to be an array, null given in /piwigo/plugins/piwigo-videojs/admin/admin_sync.php on line 78
> ```

I don't have a table "config". I have piwigo_config, however that doesn't include any vjs_sync rows

**Comment by jensbrak** [comment]
> UPDATE `config` SET `value`="a:0:{}" WHERE `param`="vjs_sync"

Well, I think it worked. Now I get timeouts instead, which I can adress. Thank you for the kind help @ipsedix ! 

> I don't have a table "config". I have piwigo_config, however that doesn't include any vjs_sync rows

The actual name of the table depend on installation, ie if any prefix to tables was chosen upon setup. I too have used the 'piwigo_' prefix so my table was named 'piwigo_config'.
If you don't have the parameter in question, I'd suggest you make sure you have the latest release of videojs (I guess you have). And if you do have the latest, I'd try to add the parameter instead of updating it. If it mess things up rather than help or do nothing, you can just remove that added row and you're back to square 1 at least.

**Comment by jensbrak** [comment]
Yup. After adressing the timeout issue, it works - confirmed. However, the timing issue is not insignificant since it seems to scale bad with large albums. But that's a completely different issue :)

**Comment by efelon** [comment]
For me the problem is fixed with the suggestion by @jensbrak to insert the value into the database, which is also called 'piwigo_config' in my case, by running the following commands in the mysql console:

```
use piwigo_db;
INSERT INTO piwigo_config(param,value) VALUES('vjs_sync','a:0:{}');
```
_'piwigo_db' is the name I choose it is most definitely different on other installations._

I had the same problem (video playback ok, sync of meta data ok, thumbnail generation nok) with the latest videojs 2.9.b and piwigo 2.10.2.

**Comment by HeroicPhoenix** [comment]
> 对我来说，问题是通过@jensbrak的建议解决的，通过在 mysql 控制台中运行以下命令，将值插入到数据库中，在我的情况下也称为“piwigo_config”：
> 
> ```
> use piwigo_db;
> INSERT INTO piwigo_config(param,value) VALUES('vjs_sync','a:0:{}');
> ```
> 
> _'piwigo_db' 是我为数据库选择的名称。它在其他安装上绝对不同。_
> 
> 我在最新的 videojs 2.9.b 和 piwigo 2.10.2 上遇到了同样的问题（视频播放正常，元数据同步正常，缩略图生成正常）。

@jensbrak is right, and my data table prefix is also piwigo_db, but the SQL statement should not be written as `insert`, it should be `update`, because 'vjs_sync' is the primary key，
The complete SQL statement is as follows
`update piwigo_config set value = 'a:0:{}' where param = 'vjs_sync';`
